javacutil/src/main/java/org/checkerframework/javacutil/TreeUtils.java

```diff
@@ -407,6 +407,30 @@ public static boolean isUseOfElement(ExpressionTree node) {
         }
     }
 
+    /**
+     * Determine whether the given MethodInvocationTree has an underlying element.
+     *
+     * @param node the MethodInvocationTree to test
+     * @return true if the tree refers to an identifier, member select, or method invocation
+     */
+    @EnsuresNonNullIf(result = true, expression = "elementFromUse(#1)")
+    @Pure
+    public static boolean isUseOfElement(MethodInvocationTree node) {
+        return isUseOfElement((ExpressionTree) node);
+    }
+
+    /**
+     * Determine whether the given NewClassTree has an underlying element.
```

Comment:
If this returns false, then is the NewClassTree definitely creating an anonymous class?  Are there any other possibilities?  I have a similar question about the MethodInvocationTree overload:  when is a method call not to a method with an element?  Does that happen when the method is non-public?  (I don't think so, but I'm wondering.)

Reply:
> when is a method call not to a method with an element? 

The only time a method call doesn't have an element is when javac issued an error for the class that contains the element.  In that case, the checker should not have continued type checking, but it currently does.  This a bug.  I'm going to open a pull request to fix that bug.

Reply:
In another pull request, I'll change `elementFromUse(MethodInvocationTree node)` and `elementFromUse(NewClassTree node)` so that they never return null.  Then the two methods I added in this pull request won't be needed, so I'm going to delete them.